### PR TITLE
NIFI-2529: added support for SSLContextService protocols in HandleHtt…

### PR DESCRIPTION
**Pull request by m-hogue**
…pRequest

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by trkurc** [comment]
adding this to my review pile for the night

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java [hunk 1]
```diff
@@ -90,10 +90,11 @@
     @WritesAttribute(attribute = "http.method", description = "The HTTP Method that was used for the request, such as GET or POST"),
     @WritesAttribute(attribute = HTTPUtils.HTTP_LOCAL_NAME, description = "IP address/hostname of the server"),
     @WritesAttribute(attribute = HTTPUtils.HTTP_PORT, description = "Listening port of the server"),
-    @WritesAttribute(attribute = "http.query.string", description = "The query string portion of hte Request URL"),
+    @WritesAttribute(attribute = "http.query.string", description = "The query string portion of the Request URL"),
     @WritesAttribute(attribute = HTTPUtils.HTTP_REMOTE_HOST, description = "The hostname of the requestor"),
     @WritesAttribute(attribute = "http.remote.addr", description = "The hostname:port combination of the requestor"),
     @WritesAttribute(attribute = "http.remote.user", description = "The username of the requestor"),
+    @WritesAttribute(attribute = "http.protocol", description = "The protocol used to communicate"),
```
Comment (trkurc) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java]: how did you decide on this attribute name? was it used in another processor? to me, it was a bit confusing, as I'd have thought maybe HTTP/1.1 or HTTP/2 would be the http protocol. would http.security.protocol make more sense? @alopresto may be able to weigh in on a preferred attribute name for this.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java [hunk 2]
```diff
@@ -446,6 +447,7 @@ private SslContextFactory createSslFactory(final SSLContextService sslService, f
 
         sslFactory.setNeedClientAuth(needClientAuth);
         sslFactory.setWantClientAuth(wantClientAuth);
+        sslFactory.setProtocol(sslService.getSslAlgorithm());
```
Comment (trkurc) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java]: I didn't run this one yet, but does it have the same issue as the ListenHTTP with starting up successfullly, but not supporting the ssl/tls choice when a request is made?

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java (hunk 1 above)
Comment (m-hogue) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java]: This actually is the `http.protocol`, so it'd be HTTP/1.1 or HTTP/2 as you noted. While potentially worthwhile, i'd admittedly added this while searching for a way to validate the SSL protocol negotiated by the client & server. I can drop the property if it's not wanted.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java (hunk 2 above)
Comment (m-hogue) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java]: Yep, the exact same problem. Once resolved on https://github.com/apache/nifi/pull/1986, i'll do the same here.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java (hunk 1 above)
Comment (trkurc) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java]: Ah, I missed this. Carry on.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java (hunk 2 above)
Comment (trkurc) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java]: Approach from #1986 looks good.

**Comment by m-hogue** [comment]
@trkurc : pushed same SSL protocol checks as added in #1985. Let me know if you'd like any additional changes.

**Comment by alopresto** [comment]
I have similar concerns with this as with [PR 1986](https://github.com/apache/nifi/pull/1986) about throwing exceptions on 71% of the available options in the protocol version dropdown.

**Comment by m-hogue** [comment]
@trkurc @alopresto : I've made changes here in line with #1986 and forced the use of `RestrictedSSLContextService` here as well. This is ready for review.

**Comment by alopresto** [comment]
@m-hogue I apologize for the delay on this. I reviewed and verified that the tests pass, the checkstyle was good, and a flow I developed using `HandleHTTPRequest` and `HandleHTTPResponse` worked as expected (succeeded with TLS v1.2 incoming connection; failed when TLSv1.1 was specified). I merged and closed the Jira. Expect this in Apache NiFi 1.4.0. Thanks for your contribution.

**Comment by m-hogue** [comment]
No worries, @alopresto. Thanks for the review!

**Pull request closed without merging**